src/libraries/System.Private.CoreLib/src/System/Enum.EnumInfo.cs

```diff
@@ -22,10 +22,7 @@ public EnumInfo(bool hasFlagsAttribute, TUnderlyingValue[] values, string[] name
                 Values = values;
                 Names = names;
 
-                if (!AreSorted(values))
-                {
-                    Array.Sort(keys: values, items: names);
-                }
+                Array.Sort(keys: values, items: names);
```

Comment:
Since already sorted should be an extremely common case, might it be worth keeping the guard check?